scripts/lib/CIME/utils.py

```diff
@@ -227,10 +227,7 @@ def run_sub_or_cmd(cmd, cmdargs, subname, subargs, logfile=None, case=None,
         logger.info("   Running {} ".format(cmd))
         if case is not None:
             case.flush()
-        stat, output, errput = run_cmd("{} {}".format(cmd, cmdargs), input_str=input_str, from_dir=from_dir,
-                                 verbose=verbose, arg_stdout=arg_stdout, arg_stderr=arg_stderr, env=env,
-                                 combine_output=combine_output)
-
+        output = run_cmd_no_fail("{} {}".format(cmd, cmdargs), combine_output=True)
```

Comment:
I like this better, although it makes most of the optional arguments to run_sub_or_cmd useless. I'd recommend removing the ones that aren't being used.

Reply:
FYI, pylint should be flagging the unused args.